### PR TITLE
docs: Update holocron usage#1

### PR DESCRIPTION
**Pull request by AdamFoskit**
This is a simple example of a basic 'hello world' holocron module.

## Description
This is an attempt to help the readability of the holocron usage #1 documentation. I debated adding childroutes so the code could work out-of-the-box for beginners/first-time users.

## Motivation and Context
When I was reading the holocron documentation for the first time, I was confused by the different '.inputs', the declared shape of the 'moduleState' vs the usage of the moduleState in the function, and an example of a reducer.

## How Has This Been Tested?
This was tested locally using holocron 1.1.0, and one-apps' npx module generator code for a 'root module' with some modifications: childRoutes removed, prop-types added, and immutable added, with a custom reducer, shouldModuleReload, and loadModuleData.

## Types of Changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation (adding or updating documentation)
- [ ] Dependency update

## Checklist:
- [x] My change requires a change to the documentation and I have updated the documentation accordingly.
- [x] My changes are in sync with the code style of this project.
- [x] There aren't any other open Pull Requests for the same issue/update.
- [ ] These changes should be applied to a maintenance branch.
- [ ] This change requires cross browser checks.
- [ ] This change impacts caching for client browsers.
- [ ] This change adds additional environment variable requirements for Holocron users.
- [ ] I have added the Apache 2.0 license header to any new files created.

## What is the Impact to Developers Using Holocron?
This change will hopefully allow beginners and first-time users that are reading the documentation to have a more clear understanding of the basic parts of holocron in a small example.

### CODE REVIEW

**Comment by AdamFoskit** [comment]
I think the npx generator's childRoutes could be added to allow the user to copy/paste the code for it to work out-of-the-box.
```jsx
import childRoutes from '../childRoutes';
```
...
```jsx
HelloWorld.childRoutes = childRoutes;
```
Two lines of code to prevent -potentially- a lot of confusion.

**Comment by JamesSingleton** [comment]
> I think the npx generator's childRoutes could be added to allow the user to copy/paste the code for it to work out-of-the-box.
> 
> ```js-jsx
> import childRoutes from '../childRoutes';
> ```
> 
> ...
> 
> ```js-jsx
> HelloWorld.childRoutes = childRoutes;
> ```
> 
> Two lines of code to prevent -potentially- a lot of confusion.

Added into the example? If so, not every module has `childRoutes`.

File: packages/holocron/docs/api/README.md [hunk 1]
```diff
@@ -103,29 +103,47 @@ The optional `holocron` object set to the parent React Component inside a Holocr
...
   loadModuleData,
   shouldModuleReload,
-  options,
+  // options,
+};
+
+if (!global.BROWSER) {
+  // eslint-disable-next-line global-require
+  HelloWorld.appConfig = require('../appConfig').default;
 }
```
Comment (JAdshead) [inline packages/holocron/docs/api/README.md:147]: `appConfig` is specific to One App. One App can't work without holocron but holocron is technically not reliant on One App. There is a lot of coupling between the two in real terms but we want to try and keep Holocron decoupled from one app.  

```suggestion
```

**Review by JAdshead — CHANGES_REQUESTED** [review]

File: packages/holocron/docs/api/README.md (hunk 1 above)
Comment (AdamFoskit) [reply packages/holocron/docs/api/README.md:147]: My fault! I will remove that.

**Comment by AdamFoskit** [comment]
> > I think the npx generator's childRoutes could be added to allow the user to copy/paste the code for it to work out-of-the-box.
> > ```
> > import childRoutes from '../childRoutes';
> > ```
> > 
> > 
> > ...
> > ```
> > HelloWorld.childRoutes = childRoutes;
> > ```
> > 
> > 
> > Two lines of code to prevent -potentially- a lot of confusion.
> 
> Added into the example? If so, not every module has `childRoutes`.

I may have some confusion on holocron vs one-app. I was under the impression all modules needed at least 1 child route, the "/" path (for itself). Maybe that is one-app specific, and not holocron specific.

**Comment by JamesSingleton** [comment]
@AdamFoskit, nope _technically_ only the root module needs `childRoutes` to load your others modules. But you could have a module that is just a widget on a screen and isn't rendered on a route itself.

**Comment by AdamFoskit** [comment]
@JamesSingleton Ah okay, that makes perfect sense, actually. Thanks for the explanation!

**Review by JAdshead — APPROVED** [review]

**Review by infoxicator — APPROVED** [review]

**Pull request merged**